x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/utils/utils.ts

```diff
@@ -126,7 +126,7 @@ export const hasReadIndexPrivileges = async (args: {
   // Some indices have read privileges others do not.
   if (indexesWithNoReadPrivileges.length > 0) {
     const indexesString = JSON.stringify(indexesWithNoReadPrivileges);
-    warningStatusMessage = `This rule may not have the required read privileges to the following index patterns: ${indexesString}`;
+    warningStatusMessage = `The concrete indices that match this index pattern may not be available to this rule's api key: ${indexesString}`;
```

Comment:
I probably miss the context, but does our clients know what is `rule's api key`?

Reply:
Agree with @yctercero about linking to the Kibana docs. The Security docs briefly mention API keys in procedures for setting up [detection features](https://www.elastic.co/guide/en/security/current/detections-permissions-section.html#alerting-auth-model) and [cross-cluster search](https://www.elastic.co/guide/en/security/current/rules-cross-cluster-search.html#update-api-key), but the kibana docs are better for getting a general idea of what API keys are and how they work.